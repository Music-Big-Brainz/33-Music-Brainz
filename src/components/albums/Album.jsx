import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


const Album = ({ name, albumId, title, date }) => (
  <Link to={`/releases/${name}/title/${albumId}/${title}`}>

    <figure key={albumId}>
      <p>{title}</p>
      <p>{date}</p>
      <img 
        src={`http://coverartarchive.org/release/${albumId}/front`}
        alt="No album art"
        style={{ width: '175px' }} />
    </figure>
  </Link>

);

Album.propTypes = {
  albumId: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
};

export default Album;
