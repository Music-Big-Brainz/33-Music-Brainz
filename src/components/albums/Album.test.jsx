import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Album from './Album';
import { MemoryRouter } from 'react-router-dom';

describe('displays an album', () => {
  afterEach(() => cleanup());
  it('renders one single album', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Album name="" albumId="" title="" date="" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
