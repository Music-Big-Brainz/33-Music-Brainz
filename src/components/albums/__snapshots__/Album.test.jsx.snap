// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an album renders one single album 1`] = `
<DocumentFragment>
  <a
    href="/releases/Prince/title/281e42d6-389e-4224-9bc6-84220a62d50f/Dirty Mind"
  >
    <figure>
      <p>
        Dirty Mind
      </p>
      <p>
        1980
      </p>
      <img
        alt="No album art"
        src="http://coverartarchive.org/release/281e42d6-389e-4224-9bc6-84220a62d50f/front"
        style="width: 175px;"
      />
    </figure>
  </a>
</DocumentFragment>
`;
